### PR TITLE
feat: EB-27262 Add log for request/response

### PR DESCRIPTION
**Pull request by aallier**

### CODE REVIEW

**Review by Shad-Z — APPROVED** [review]

File: EventSubscriber/LogTrait.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    public function isContentExceedMaxSize(string $content): bool
+    {
+        return mb_strlen($content) > self::CONTENT_MAX_SIZE;
+    }
+
+    public function splitContent(string $content): string
+    {
+        return mb_substr($content, 0, self::CONTENT_MAX_SIZE / 2) .
+            self::SPLIT_SEPARATOR .
+            mb_substr($content, -self::CONTENT_MAX_SIZE / 2);
```
Comment (ch3ric) [inline EventSubscriber/LogTrait.php:21]: what's the point of splitting? the second part after ... should be removed completly?

File: EventSubscriber/LogRequestSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                'method' => $request->getMethod(),
+                'request_uri' => $request->getRequestUri(),
+                'headers' => $request->headers->all(),
+                'query' => $request->query->all(),
+                'request' => $request->request->all(),
+                'content' => $this->splitContent($request->getContent()),
+                'memory' => memory_get_usage(),
+            ],
+        );
+    }
```
Comment (ch3ric) [inline EventSubscriber/LogRequestSubscriber.php:62]: no need to duplicate everything just to change the message?

```suggestion
        $this->logger->warning(
           $isContentExceed ? 'Start request (long request content)' : 'Start request',
            [
                'method' => $request->getMethod(),
                'request_uri' => $request->getRequestUri(),
                'headers' => $request->headers->all(),
                'query' => $request->query->all(),
                'request' => $request->request->all(),
                'content' => $this->splitContent($request->getContent()),
                'memory' => memory_get_usage(),
            ],
        );
    }
```

File: EventSubscriber/LogTrait.php (hunk 1 above)
Comment (aallier) [reply EventSubscriber/LogTrait.php:21]: end of content can be important, I prefer truncate in the middle of string

File: EventSubscriber/LogRequestSubscriber.php (hunk 2 above)
Comment (aallier) [reply EventSubscriber/LogRequestSubscriber.php:62]: if string < max I want log level info and if string > max I want warning, I don't want to use dynamique call (perf issue)

**Review by ch3ric — APPROVED** [review]

**Pull request merged**